invoker/core/src/main/java/com/google/cloud/functions/invoker/http/HttpRequestImpl.java

```diff
@@ -120,6 +120,11 @@ public Map<String, List<String>> getHeaders() {
         .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
Kind of a drive-by comment here. I wrote this code originally and I admit the case-sensitivity aspect didn't occur to me. An alternative fix would be to change this method like this:

```java
return Collections.list(request.getHeaderNames()).stream()
    .collect(
        toMap(
            name -> name,
            name -> Collections.list(request.getHeaders(name)),
            (a, b) -> b,
            () -> new TreeMap<>(String.CASE_INSENSITIVE_ORDER));
```

That also has the advantage of a predictable iteration order, and it means the `getHeaders()` map is also case-insensitive, not just `getFirstHeader`.

Regardless of whether you choose this alternative, I think it would be good to remove the `.map` step in favour of doing the transformation in `.toMap`. Also, even though `HttpResponse.getFirstHeader` is presumably called much less often, I think it should also be case-insensitive.

Reply:
Thanks for jumping in on this, I like the solution of moving to TreeMap. I've gone ahead and pulled your code snippet into the PR and updated the HttpResponseImpl definitions. 

WDYT?